libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -515,7 +516,7 @@ void AC_PosControl::init_xy_controller()
     init_ekf_xy_reset();
 
     // initialise z_controller time out
-    _last_update_xy_us = AP_HAL::micros64();
+    _last_update_xy_us = AP::ins().get_last_update_usec();
```

Comment:
Why the switch to `last_update` here? I guess pos control cares about the time where as attitude control only cares about the change in time? There could be some case where the change in last update does not match the dt, at least micros should (ish).

Reply:
This is where we are checking that we have initialised the position controller on the last loop or we were running on the last loop. This is why this is different.

Reply:
Do you think we want this or _last_sample_us?

Reply:
This is effectivly the replacement of the flow of control error. We just want to make sure we have initialised the Pos controller. So strictly speeking we just need to make sure that AP::ins().get_last_update_usec() - _last_update_z_us = get_dt()* 1000000.0 but I doubled it to give it a buffer.